File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        
        int ans = 0;
        int leftmin = prices[0];
        
        
        for(int i = 1;i<prices.size();i++){
            
            if(prices[i]<leftmin) leftmin = prices[i];
            else
                ans = max(prices[i]-leftmin,ans);
            
        }
        
        return ans;
        
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        ans=0
        leftmin = prices[0]
        
        for i in range(1,len(prices)):
            
            if prices[i]<leftmin:
                leftmin = prices[i]
            else:
                ans=max(ans,prices[i]-leftmin)
        
        return ans
            
            
            
        